spec/features/admin/invoice_print_spec.rb

```diff
@@ -23,6 +23,10 @@
                                                 payment_state: 'balance_due')
   end
 
+  before do
+    stub_request(:get, ->(uri) { uri.to_s.include? "/css/mail" })
```

Comment:
Why do we need to stub this? Shouldn't it be delivered by the app?

Reply:
There was some weirdness here related to the way emails load CSS and how the test suite handles URLs. It ends up with the CSS being included with `http://test.com/whatever.css`, which then gets treated as an external resource in this spec, and the request fails because it's not actually a valid URL.

Reply:
Ah... and we're also rendering a PDF as part of this test but the template is re-using an email template? And then we're doing something weird where the PDF then gets converted back into a page with `#convert_pdf_to_page` so we can see it's contents. It's pretty funky!

Reply:
Funky indeed! Perhaps a bit out of scope here but this is the point I've been stuck on [this PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/8434) for a while now.

Reply:
Aha! Yeah that makes sense, I think we need to look at that PDF generation and it's specs and improve them a bit...